packages/oidc-dpop-client-browser/src/dpop/dpop.spec.ts

```diff
@@ -48,3 +57,94 @@ describe("generateRsaKey", () => {
     expect(key.kty).toEqual("RSA");
   });
 });
+
+describe("signJWT/decodeJWT", () => {
+  it("generates a JWT that can be decoded without signature verification", async () => {
+    const key = await generateKeyForDpop();
+    const payload = { testClaim: "testValue" };
+    const jwt = await signJWT(payload, key, {
+      algorithm: "RS256",
+    });
+    const decoded = await decodeJWT(jwt);
+    expect(decoded.testClaim).toEqual(payload.testClaim);
+  });
+
+  it("can verify the ES256 signature of the generated JWT", async () => {
```

Comment:
What is it supposed to do when the verification fails? Is it doable to have a test for that?

Reply:
If the verification fails, it throws. I added a test there : d855d58d7e5e8871e800597275e6a77844d6ea3f. That's the behaviour I'd expect for a low-level library like this, but one layer higher, when requesting a DPoP token (bound to the key) to the IdP, this exception should be catched, at least to add more context.